.github/workflows/main.yml

```diff
@@ -0,0 +1,425 @@
+            lua_include: "/usr/include/luajit-2.1"
+            luarocks_args: "--lua-suffix=jit-2.1.0-beta3"
+
+          # Lua 5.2 with older lgi and screen size not divisible by 2.
+          - test_name: "oldlgi-lua5.2"
+            lua_version: "5.2"
+            lua_name: "lua5.2"
+            lgi_version: "0.8.0"
+            tests_screen_size: "1921x1079"
+            check_qa: true
+            empty_theme_while_loading: true
+            # SOURCE_DATE_EPOCH: used for stable dates during build.
+            source_date_epoch: "1893456000"
+            test_prev_commits: true
+
+    env:
+      LUA: ${{ matrix.lua_version }}
+      LUAINCLUDE: /usr/include/lua${{ matrix.lua_version }}
+      LUALIBRARY: /usr/lib/x86_64-linux-gnu/liblua${{ matrix.lua_version }}.so
+      TESTS_SCREEN_SIZE: ${{ matrix.tests_screen_size || '' }}
+      SOURCE_DATE_EPOCH: ${{ matrix.source_date_epoch || '' }}
+      TEST_TIMEOUT: '100'
+      CI: 'true'
+
+    steps:
+      # Create a cache invalidation key based on the current year + week.
+      # This way, packages will be checked for updates once every week.
+      - name: Get Date
+        id: get-date
+        run: echo "::set-output name=date::$(/bin/date -u "+%Y%W")"
+
+      - name: Create apt cache directory
+        run: mkdir /tmp/apt
+
+      - name: Cache apt packages
+        id: cache-apt
+        uses: actions/cache@v2
+        with:
+          path: /tmp/apt/*.deb
+          # The trailing number serves as a version flag that can be incremented
+          # to invalidate the cache after changing the list of packages.
+          key: ${{ runner.os }}-${{ steps.get-date.outputs.date }}-apt-3
+
+      - name: Download apt packages
+        if: steps.cache-apt.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get update
+          sudo apt-get install --download-only -y --no-install-recommends \
+            asciidoctor \
+            cmake \
+            dbus-x11 \
+            gettext \
+            gir1.2-gtk-3.0 \
+            gir1.2-pango-1.0 \
+            git \
+            libdbus-1-dev \
+            libgirepository1.0-dev \
+            libnotify-bin \
+            libpango1.0-dev \
+            libstartup-notification0-dev \
+            libx11-xcb-dev \
+            libxcb-cursor-dev \
+            libxcb-icccm4-dev \
+            libxcb-keysyms1-dev \
+            libxcb-randr0-dev \
+            libxcb-shape0-dev \
+            libxcb-util0-dev \
+            libxcb-xfixes0-dev \
+            libxcb-xinerama0-dev \
+            libxcb-xkb-dev \
+            libxcb-xrm-dev \
+            libxcb-xtest0-dev \
+            libxdg-basedir-dev \
+            libxkbcommon-dev \
+            libxkbcommon-x11-dev \
+            wmctrl \
+            x11-apps \
+            xcb-proto \
+            xdotool \
+            xorg \
+            xserver-xephyr \
+            xterm \
+            xutils-dev \
+            xvfb \
+            zsh \
```

Comment:
Any guesses when zsh is used in the builds ?

Reply:
Not in the build, but when testing `lib/awful/completion.lua`: https://github.com/awesomeWM/awesome/blob/aeb2d85dad31073eaf979201f35dece4a51ccd91/spec/awful/completion_spec.lua